lib/runtime/src/component/endpoint.rs

```diff
@@ -229,3 +337,66 @@ impl EndpointConfigBuilder {
         Ok(())
     }
 }
+
+/// Context for building transport type - determines port and formatting differences
+enum TransportContext {
+    /// For health check targets
+    HealthCheck,
+    /// For discovery service registration
+    Discovery,
+}
+
+/// Build transport type based on request plane mode and context
+///
+/// This unified function handles both health check and discovery transport building,
+/// with context-specific differences:
+/// - HTTP: Both use the same port (default 8888, configurable via DYN_HTTP_RPC_PORT)
+/// - TCP: Health check omits endpoint suffix, discovery includes it for routing
+/// - NATS: Identical for both contexts
+fn build_transport_type(
+    mode: RequestPlaneMode,
+    endpoint_name: &str,
+    subject: &str,
+    context: TransportContext,
+) -> TransportType {
+    match mode {
+        RequestPlaneMode::Http => {
+            let http_host = crate::utils::get_http_rpc_host_from_env();
+            // Both health check and discovery use the same port (8888) where the HTTP server binds
+            let http_port = std::env::var("DYN_HTTP_RPC_PORT")
+                .ok()
+                .and_then(|p| p.parse::<u16>().ok())
+                .unwrap_or(8888);
+            let rpc_root =
+                std::env::var("DYN_HTTP_RPC_ROOT_PATH").unwrap_or_else(|_| "/v1/rpc".to_string());
+
+            let http_endpoint = format!(
+                "http://{}:{}{}/{}",
+                http_host, http_port, rpc_root, endpoint_name
+            );
+
+            TransportType::Http(http_endpoint)
+        }
+        RequestPlaneMode::Tcp => {
+            let tcp_host = crate::utils::get_tcp_rpc_host_from_env();
+            let tcp_port = std::env::var("DYN_TCP_RPC_PORT")
+                .ok()
+                .and_then(|p| p.parse::<u16>().ok())
+                .unwrap_or(9999);
+
+            let tcp_endpoint = match context {
+                TransportContext::HealthCheck => {
```

Comment:
not sure I follow here - wouldn't health check also need routing? we associate the health check and payload per endpoint -